server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/copybooks/analysis/AbstractCopybookAnalysis.java

```diff
@@ -90,6 +92,7 @@ public PreprocessorFunctor handleCopybook(
                       .name(retrieveCopybookName(copySource))
                       .context(context)
                       .documentUri(documentUri)
+                      .flavor(Optional.ofNullable(flavorType).map(Enum::name).orElse(null))
```

Comment:
What about introducing an enum element for the non-dialect calls to avoid dealing with null? You may even provide the string that should be passed to the client as a field inside the enum, and set it via constructor calls.